pyxem/signals/segments.py

```diff
@@ -0,0 +1,444 @@
+            factors_of_segments = np.delete(
+                factors_of_segments, delete_indices, axis=0)
+
+        try:
+            segments = Signal2D(segments).transpose(navigation_axes=[0],
+                                                    signal_axes=[2, 1])
+        except TraitError:
+            if segments.shape[0] == 0:
+                raise ValueError('No segments were found. Check the input '
+                                 'parameters.')
+        factors_of_segments = Signal2D(factors_of_segments)
+        learning_segment = LearningSegment(segments, factors_of_segments)
+        return learning_segment
+
+
+class VDFSegment:
+    _signal_type = "vdf_segment"
+
+    def __init__(self, segments, vectors_of_segments, intensities=None):
+        # Segments as Signal2D
+        self.segments = segments
+        # DiffractionVectors
+        self.vectors_of_segments = vectors_of_segments
+        # Intensities corresponding to each vector
+        self.intensities = intensities
+
+    def correlate_vdf_segments(self, corr_threshold=0.7, vector_threshold=4,
+                               segment_threshold=3):
+        """Iterates through VDF segments and sums those that are
+        associated with the same segment. Summation will be done for
+        those segments that have a normalised cross correlation above
+        corr_threshold. The vectors of each segment sum will be updated
+        accordingly, so that the vectors of each resulting segment sum
+        are all the vectors of the original individual segments. Each
+        vector is assigned an intensity that is the integrated intensity
+        of the segment it originates from.
+
+        Parameters
+        ----------
+        corr_threshold : float
+            Segments will be summed if they have a normalized cross-
+            correlation above corr_threshold. Must be between 0 and 1.
+        vector_threshold : int, optional
+            Correlated segments having a number of vectors less than
+            vector_threshold will be discarded.
+        segment_threshold : int, optional
+            Correlated segment intensities that lie in a region where
+            a number of segments less than segment_thresholdhave been
+            found, are set to 0, i.e. the resulting segment will only
+            have intensities above 0 where at least a number of
+            segment_threshold segments have intensitives above 0.
+
+        Returns
+        -------
+        vdfseg : VDFSegment
+            The VDFSegment instance updated according to the image
+            correlation results.
+        """
+        vectors = self.vectors_of_segments.data
+        if len(np.shape(vectors)) <= 1:
+            raise ValueError("Input vectors are not of correct shape. Try to "
+                             "rerun correlate_segments on original VDFSegment "
+                             "resulting from get_vdf_segments.")
+        if segment_threshold > vector_threshold:
```

Comment:
This condition needs a test.

Reply:
Will write test.

Reply:
Now tested c/o https://github.com/dnjohnstone/pyxem/pull/5